fractal_server/app/models/job.py

```diff
@@ -87,35 +86,56 @@ class Config:
         arbitrary_types_allowed = True
 
     id: Optional[int] = Field(default=None, primary_key=True)
-    project_id: int = Field(foreign_key="project.id")
-    input_dataset_id: int = Field(foreign_key="dataset.id")
-    output_dataset_id: int = Field(foreign_key="dataset.id")
-    workflow_id: int = Field(foreign_key="workflow.id")
-    working_dir: Optional[str]
-    working_dir_user: Optional[str]
-    first_task_index: int
-    last_task_index: int
 
+    project_id: Optional[int] = Field(foreign_key="project.id")
+    project: Optional["Project"] = Relationship(  # noqa: F821
+        back_populates="job_list"
+    )
+
+    workflow_id: Optional[int] = Field(foreign_key="workflow.id")
+    workflow: Optional["Workflow"] = Relationship(  # noqa: F821
```

Comment:
Why can we import Dataset but not Workflow? Unless there are special reasons, let's keep it homogeneous.

Reply:
Actually we may even remove this relationship, if this does not create other problems.